sparseconfig.h

```diff
@@ -0,0 +1,46 @@
+/*
```

Comment:
why?

Reply:
This file is auto generated by cmake (can't say I 100% know how), so bazel need to get this from somewhere.

Reply:
Its generated by cmake of a thirdparty lib during configure phase. You need to run configure, make, install of the fetched repo and you will get this file. Its wrong adding it to gaia
 If you can not solve it, submit your pr without those files and i will try to fix it

Reply:
It's possible to fix it with `genrule`, I was trying to avoid cmake/configure in bazel as much as possible, but as you wish.

Reply:
@romange I still need to check it out, I'm trying to check if it's possible to generate this config without running cmake (but just configure maybe?), I will fix it in another PR.

Reply:
In that case submit what you can here but please do not add c++ code or change c++ code in this PR. Once you have something  basic we can look at other repos and see how they build 3rd party with bazel.